ledger/eval.go

```diff
@@ -768,19 +774,19 @@ func (eval *BlockEvaluator) transaction(txn transactions.SignedTxn, evalParams *
 		return fmt.Errorf("transaction %v: %v", txid, err)
 	}
 
-	// Validate applyData if we are validating an existing block.
-	// If we are validating and generating, we have no ApplyData yet.
-	if eval.validate && !eval.generate {
-		if eval.proto.ApplyData {
-			if !ad.Equal(applyData) {
-				return fmt.Errorf("transaction %v: applyData mismatch: %v != %v", txid, ad, applyData)
-			}
-		} else {
-			if !ad.Equal(transactions.ApplyData{}) {
-				return fmt.Errorf("transaction %v: applyData not supported", txid)
-			}
-		}
-	}
+	//// Validate applyData if we are validating an existing block.
```

Comment:
With your change, you've dropped the usage of the input parameter ad. If this is the case, remove it from the caller and clean up the code.